File: rust/session05/fizzbuzz.py
#!/usr/bin/env python

if __name__ == "__main__":
    # not possible as lambda
    def fizzbuzz(x):
        if x % 15 == 0:
            print("FizzBuzz")
        elif x % 3 == 0:
            print("Fizz")
        elif x % 5 == 0:
            print("Buzz")
        else:
            print(x)

    for n in range(1, 16):
        fizzbuzz(n)

    print("---")
    list(map(fizzbuzz, range(1, 16)))

File: rust/session05/fizzbuzz.rs
fn main() {
    // closure
    let fizzbuzz = |x| {
        if x % 15 == 0 {
            println!("FizzBuzz");
        } else if x % 3 == 0 {
            println!("Fizz");
        } else if x % 5 == 0 {
            println!("Buzz");
        } else {
            println!("{}", x);
        }
    };

    // function
    fn fizzbuzz(x: i32) {
        if x % 15 == 0 {
            println!("FizzBuzz");
        } else if x % 3 == 0 {
            println!("Fizz");
        } else if x % 5 == 0 {
            println!("Buzz");
        } else {
            println!("{}", x);
        }
    }

    for n in 1..16 {
        fizzbuzz(n)
    }

    println!("---");
    (1..16).into_iter().for_each(fizzbuzz); // pass function fizzbuzz as arg
}
